### PR TITLE
Upgrade to creating Python3 Projects

### LINKED ISSUE #14: Have Yurt Setup Django Projects That Use Python 3

### LINKED ISSUE #24: Make Yurt play nice with Ansible 2
It's the <del>future</del> present. Pay special attention to how it renders the `postactivate` script and handles `django_environments`.

### PR DESCRIPTION
**Pull request by paolopaolopaolo**
- Also, updating Ansible to be current
- Use pip3 instead of regular pip
- Many, many tweaks to get this to run all the way
- Moar tweaks: fixing project names with "-"
- No special chars in database passwords
- Refactor getting project name from git repo
- Potentially fix virtualenvwrapper stuff (make it truly install globally with `sudo`)

@baylee, @rmutter: Review please! I was planning on introducing fixes to make Yurt more flexible with systems that are using pyenv, but I figured it would probably be best to put that in another PR later.

Fixes #24, Fixes #14

### CODE REVIEW

**Comment by baylee** [comment]
Yeah, I'm not sure we would need pyenv. All the python3 stuff will always be happening within a VM, and it's pretty uncommon we have more than one project running on the same machine (I think we only do it for the yeti site?)

**Comment by paolopaolopaolo** [comment]
By "using pyenv" I meant people whose local setups on their Macs use pyenv, not an option for putting pyenv on remote or vagrant servers. It would be nice for Yurt to not break before it even hits the starting line for any python setup.

**Comment by baylee** [comment]
So you mean, my computer?

**Comment by paolopaolopaolo** [comment]
Yup! :grin: If you'd like to test this new version out, feel free!

**Comment by baylee** [comment]
One thing that seems a little odd to me is the virtualenv stuff. In python3, virtualenv is built in (as `venv`), so not sure what you're doing with `virtualenv` and `virtualenvwrapper`. 

Additionally, `pyenv` has it's own `pyenv-virtualenv` (and wrapper :stuck_out_tongue: ). I can talk to you about using these, though I haven't loved them and they don't make complete sense to me.

**Comment by paolopaolopaolo** [comment]
So I've been enforcing the use of `virtualenvwrapper` in this project for two reasons:
- adding a `postactivate` file to the `bin` of the virtualenv enables one to add virtualenv-specific environment variables
- a call to `workon <name>` is way easier to make a context manager out of than  a call to `os.path.expanduser("~/path/to/venv/bin/activate")` 
- I don't like the idea of making Ansible a global python package since it's web deployment specific

The fact that `venv` is built-in to Python3 is fabulous.

File: fabfile/setup.py [hunk 1]
```diff
@@ -39,13 +39,16 @@ def create_ansible_env():
...
     with bash():
         with settings(warn_only=True):
             local('mkvirtualenv ansible')
     with ansible():
         with settings(warn_only=True):
-            local('pip install ansible==1.9.4')
-            local('sudo ansible-galaxy install -r $FAB_PATH/../orchestration/roles/roles.txt')
+            local('pip install ansible')
+        with settings(warn_only=True):
+            local('sudo ansible-galaxy remove nodesource.node')
```
Comment (rmutter) [inline fabfile/setup.py]: Could this have some unintended consequences on removing the role for another project that's using it? Basically could this break the deployment of another app that's maybe depending on an older version of node or this role?

File: orchestration/roles/app/tasks/setup_django_app.yml [hunk 2]
```diff
@@ -1,7 +1,10 @@
 ---
 
 - name: Install packages required by the Django app inside virtualenv
-  pip: virtualenv={{ virtualenv_path }} requirements={{ requirements_file }}
+  pip: virtualenv={{ virtualenv_path }} requirements={{ requirements_file }} executable="{{ virtualenv_path }}/bin/pip3"
+
+- name: Make `manage.py` executable
+  file: path="{{project_path}}/manage.py" mode="u+x,g+x"
```
Comment (rmutter) [inline orchestration/roles/app/tasks/setup_django_app.yml]: should we keep consistency with spaces between the curly braces and the variable here and elsewhere you added them?

File: fabfile/setup.py (hunk 1 above)
Comment (paolopaolopaolo) [inline fabfile/setup.py]: `ansible-galaxy` doesn't have a set way of updating packages (which caused nodesource to break whenever they updated the node version). To truly do an update, we'd need to remove and then re-install roles, which is what this segment of the code does.

**Comment by rmutter** [comment]
@paolopaolopaolo is this ready to be merged now?

**Pull request merged**